src/libinjector/linux/methods/linux_shellcode.c

```diff
@@ -0,0 +1,164 @@
+#include "linux_shellcode.h"
+#include "linux_debug.h"
+#include "linux_syscalls.h"
+
+bool setup_mmap_trap(drakvuf_t drakvuf, drakvuf_trap_info_t* info);
+bool write_shellcode_to_mmap_location(drakvuf_t drakvuf, drakvuf_trap_info_t* info);
+
+event_response_t handle_shellcode(drakvuf_t drakvuf, drakvuf_trap_info_t* info)
```

Comment:
There is a lot happening in this function. Can you add some comment that describes the whole process?

Reply:
where should I add the comment, inside the function or near the prototype?

Reply:
```
/**
 * Comment body.
 * 
 */
 void foo() {
 ```

Reply:
Done